packages/service-library/src/servicelib/long_running_tasks/task.py

```diff
@@ -169,82 +226,135 @@ async def _stale_tasks_monitor_worker(self) -> None:
             _logger.warning(
                 "Removing stale task '%s' with status '%s'",
                 task_id,
-                self.get_task_status(task_id, with_task_context=None).model_dump_json(),
+                (
+                    await self.get_task_status(task_id, with_task_context=task_context)
+                ).model_dump_json(),
             )
             await self.remove_task(
-                task_id, with_task_context=None, reraise_errors=False
+                task_id, with_task_context=task_context, reraise_errors=False
             )
 
-    def list_tasks(self, with_task_context: TaskContext | None) -> list[TaskBase]:
+    async def _cancelled_tasks_removal_worker(self) -> None:
```

Comment:
If you abstract every worker task separately, you can move away to the corresponding class this cdoe as well

Reply:
commented above, will not do here